sw/airborne/modules/uavcan/uavcan_allocator.c

```diff
@@ -0,0 +1,269 @@
+
+#include "uavcan/uavcan.h"
+#include "uavcan/uavcan_allocator.h"
+#include "uavcan.protocol.dynamic_node_id.Allocation.h"
+#include "uavcan.protocol.GetNodeInfo.h"
+#include "uavcan.equipment.air_data.AngleOfAttack.h"
+#include "mcu_periph/gpio.h"
+
+
+/**
+ *  Dynamic node ID allocation.
+ *  See https://dronecan.github.io/Specification/6._Application_level_functions/#dynamic-node-id-allocation
+ */
+
+
+#ifndef UAVCAN_DYN_NODE_NB
+#define UAVCAN_DYN_NODE_NB 10
```

Comment:
Maybe name it UAVCAN_MAX_NODES, since it also contains the nodes of the fixed ids? And by default make it a bit higher, since we are usually way above this number. So for us 50 would make more sense.